### ISSUE TITLE
VCPKG_FORCE_SYSTEM_BINARIES needs to be set during vcpkg bootstrap

### ISSUE THREAD

**Issue by kdowney-lot49** [msclock/features#15]
On a Mac M1 (arm64) the feature build fails because VCPKG_FORCE_SYSTEM_BINARIES has not been set.

This one-line change fixed it:

```
VCPKG_FORCE_SYSTEM_BINARIES=1 "${VCPKG_ROOT}"/bootstrap-vcpkg.sh
```

which I brought into my dev container locally as a workaround, but it would be nice to have it fixed at source.

**Comment by msclock** [comment]
> On a Mac M1 (arm64) the feature build fails because VCPKG_FORCE_SYSTEM_BINARIES has not been set.
> 
> This one-line change fixed it:
> 
> ```
> VCPKG_FORCE_SYSTEM_BINARIES=1 "${VCPKG_ROOT}"/bootstrap-vcpkg.sh
> ```
> 
> which I brought into my dev container locally as a workaround, but it would be nice to have it fixed at source.

@0xg0nz0 Thanks for your suggestion. I will dig it and seek for a solution later.

**Comment by kdowney-lot49** [comment]
@msclock this appears to be still a problem, but only for vcpkg-test linking (is that even needed?):

```
  [2024-04-20T14:57:03.427Z] #42 1434.2 [216/216] Linking CXX executable vcpkg-test
  [2024-04-20T14:57:03.535Z] #42 1434.3 Environment variable VCPKG_FORCE_SYSTEM_BINARIES must be set on arm, s390x, ppc64le and riscv platforms.
```

FYI, full failed CI build when I remove my workaround to export VCPKG_FORCE_SYSTEM_BINARIES=1:

https://github.com/iggy-rs/iggy-cpp-client/actions/runs/8765694062/job/24056959525

**Comment by kdowney-lot49** [comment]
Reading the relevant scripts, I have a hunch this line in `main.sh` is the issue:

```
"${VCPKG_ROOT}"/bootstrap-vcpkg.sh
```

The failure is happening during the build of the vcpkg binary, and it looks like you added VCPKG_FORCE_SYSTEM_BINARIES=1 in your updaterc function but not here. It may be that this has not yet taken effect by the time you call `bootstrap-vcpkg.sh`. Maybe try this?

```
VCPKG_FORCE_SYSTEM_BINARIES=1 "${VCPKG_ROOT}"/bootstrap-vcpkg.sh
```

**Comment by msclock** [comment]
@0xg0nz0 Ah, I have missed it. And now it works after #21 and #22 merged.

**Comment by kdowney-lot49** [comment]
Thanks @msclock ! You can close this, I can confirm it's working my end with the [latest dev container build](https://github.com/iggy-rs/iggy-cpp-client/actions/runs/8783593095). Thanks for contributing this feature!

**Comment by msclock** [comment]
you're welcome:)
